pkg/deployments/src/artifact.ts

```diff
@@ -58,6 +74,28 @@ export function checkArtifact(task: Task): void {
   }
 }
 
+/**
+ * Ensure there is a readme with the expected link to the artifact file (and no invalid ones).
+ */
+function checkReadmeArtifactLink(task: Task, contractName: string, readmeLines: string[]): void {
+  const expectedContent = `- [\`${contractName}\` artifact](./artifact/${contractName}.json)`;
+  let found = false;
+
+  for (const line of readmeLines) {
+    // Tolerate differences in case (e.g., veBoost vs VeBoost).
+    if (line.toLowerCase() == expectedContent.toLowerCase()) {
+      found = true;
+      break;
+    }
+  }
+
+  if (found) {
```

Comment:
```suggestion
  const linkFound = readmeLines.find((line) => line.toLowerCase() == expectedContent.toLowerCase()) !== undefined;

  if (linkFound) {
```

Slightly more compact version